### ISSUE TITLE
Custom path for loading monaco doesn't work

### ISSUE THREAD

**Issue by Hkattelu** [leolorenzoluis/xyz.MonacoEditorLoader#63]
\I spent some time trying to pass in a custom path for the *loadMonacoEditor directive and I realized that no matter what I changed the path to, the path the service was trying to load from would not change.

I noticed that the editor load logic actually happens in the constructor of the loader service:
https://github.com/leolorenzoluis/xyz.MonacoEditorLoader/blob/55c9ad2b12e808eabaf39b3b88105352ced21c58/lib/monaco-editor-loader/monaco-editor-loader.service.ts#L15

Even though the directive has an `@Input` that passes in the custom path, that doesn't actually ever activate since the loading logic executes in the constructor. By the time the input is set, the LoaderService has already attempted to load the editor.

Additionally, the README suggests that it's the inclusion of the *loadMonacoEditor directive
that loads the editor, but this isn't the case.

```
<monaco-editor *loadMonacoEditor></monaco-editor>
```

Even without the directive, the editor will load since the logic will be execute on construction of the service when any component imports the MonacoEditorLoaderModule.

I propose that the editor service should have a load() method that is actually only called in the MonacoEditorLoaderDirective, after the `@Input` are set within the ngOnInit.

**Comment by leolorenzoluis** [comment]
How are you setting the custom path? I tested with custom path and it is working as expected. The life cycle is correct. When the input directive is set, the property is respected to whatever value you set.

**Comment by Hkattelu** [comment]
@leolorenzoluis Thanks for looking into this. I set the directive as specified in the README

```
<monaco-editor *loadMonacoEditor="'test/monaco'"></monaco-editor>
```

Perhaps there is a race condition. In your test, is the AMD loader being loaded via loaderScript? If so, onGotAmdLoader() might be executing after the Input is set - but that wouldn't be the case if the window.require was provided and onGotAmdLoader() fires immediately

**Comment by leolorenzoluis** [comment]
Ah makes sense now. I have my require as undefined. Is this what you propose?

```
import { Injectable, NgZone } from '@angular/core';
// tslint:disable-next-line:import-blacklist
import { BehaviorSubject } from 'rxjs';

@Injectable()
export class MonacoEditorLoaderService {
    isMonacoLoaded: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
    private _monacoPath = 'assets/monaco-editor/vs';
    set monacoPath(value: any) {
        if (value) {
            this._monacoPath = value;
            this.load();
        }
    }

    load = () => {
            // Load monaco
            console.log(this._monacoPath);
            (<any>window).require.config({ paths: { 'vs': this._monacoPath } });
            (<any>window).require(['vs/editor/editor.main'], () => {
                this.ngZone.run(() => this.isMonacoLoaded.next(true));
            });
    };

    constructor(private ngZone: NgZone) {
        // Load AMD loader if necessary
        if (!(<any>window).require) {
            const loaderScript = document.createElement('script');
            loaderScript.type = 'text/javascript';
            loaderScript.src = `${this._monacoPath}/loader.js`;
            loaderScript.addEventListener('load', this.load);
            document.body.appendChild(loaderScript);
        }
    }
}
```

**Comment by Hkattelu** [comment]
Yes, that should do the job. Thanks!

**Comment by leolorenzoluis** [comment]
@Hkattelu Published. Should be fixed on `8.0.13`
